File: SDKDemo/SDKDemo/BannerAdViewController.m
//
//  BannerAdViewController.m
//  SDKDemo
//
//  Created by 马秉尧 on 2019/9/25.
//  Copyright © 2019 新义互联（北京）科技有限公司. All rights reserved.
//

@import AdtalosSDK;

#import "BannerAdViewController.h"

@interface BannerAdViewController ()<AdtalosListener>

- (void) onAdRendered;
- (void) onAdImpressionFinished;
- (void) onAdImpressionReceivedError:(NSError *)error;
- (void) onAdLoaded;
- (void) onAdFailedToLoad:(NSError *)error;
- (void) onAdClicked;
- (void) onAdLeftApplication;
- (void) onAdOpened;
- (void) onAdClosed;

@end

@implementation BannerAdViewController {
    AdtalosAdView *bannerAdView;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    bannerAdView = [[AdtalosAdView alloc] initWithFrame:CGRectMakeWithAdtalosBannerAdSize(0, 50)];
    bannerAdView.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin;
    [bannerAdView loadAd: @"209A03F87BA3B4EB82BEC9E5F8B41383"];
    bannerAdView.delegate = self;
    [self.view addSubview:bannerAdView];
    
    UIButton *closeButton = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    CGFloat x = ([UIScreen mainScreen].bounds.size.width - 100) / 2;
    CGFloat y = [UIScreen mainScreen].bounds.size.height - 100;
    closeButton.frame = CGRectMake(x, y, 100, 50);
    closeButton.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin;
    [closeButton setTitle:@"关闭" forState:UIControlStateNormal];
    [closeButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [closeButton setBackgroundColor:[UIColor redColor]];
    [closeButton addTarget:self action:@selector(closeButtonClick:) forControlEvents:UIControlEventTouchUpInside];
    [closeButton.layer setMasksToBounds:YES];
    [closeButton.layer setCornerRadius:10.0];
    [self.view addSubview:closeButton];
}

- (void)closeButtonClick:(UIButton *)button {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void) onAdRendered {
    NSLog(@"Adtalos onAdRendered");
}

- (void) onAdImpressionFinished {
    NSLog(@"Adtalos onAdImpressionFinished");
}

- (void) onAdImpressionReceivedError:(NSError *)error {
    NSLog(@"Adtalos onAdImpressionReceivedError: %@", error);
}

- (void) onAdLoaded {
    NSLog(@"Adtalos onAdLoaded");
}

- (void) onAdFailedToLoad:(NSError *)error {
    NSLog(@"Adtalos onAdFailedToLoad: %@", error);
}

- (void) onAdClicked {
    NSLog(@"Adtalos onAdClicked");

}

- (void) onAdLeftApplication {
    NSLog(@"Adtalos onAdLeftApplication");
}

- (void) onAdOpened {
    NSLog(@"Adtalos onAdOpened");
}

- (void) onAdClosed {
    NSLog(@"Adtalos onAdClosed");
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    bannerAdView = nil;
    // Dispose of any resources that can be recreated.
}

@end

File: SDKSwiftDemo/SDKSwiftDemo/BannerAdViewController.swift
//
//  BannerAdViewController.swift
//  SDKSwiftDemo
//
//  Created by 马秉尧 on 2019/10/17.
//  Copyright © 2019 新义互联（北京）科技有限公司. All rights reserved.
//

import Foundation

import UIKit
import AdtalosSDK

class BannerAdViewController: UIViewController, AdtalosListener {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = UIColor.white
        let bannerAdView: AdtalosAdView = AdtalosAdView(frame: CGRectMakeWithAdtalosBannerAdSize(0, 50))
        bannerAdView.autoresizingMask = [.flexibleLeftMargin, .flexibleRightMargin]
        bannerAdView.loadAd("209A03F87BA3B4EB82BEC9E5F8B41383")
        bannerAdView.delegate = self
        view.addSubview(bannerAdView)

        let closeButton: UIButton = UIButton(type: .roundedRect)
        let x: CGFloat = (UIScreen.main.bounds.size.width - 100) / 2
        let y: CGFloat = (UIScreen.main.bounds.size.height - 100)
        closeButton.frame = CGRect(x: x, y: y, width:100, height:50)
        closeButton.autoresizingMask = [.flexibleWidth, .flexibleHeight, .flexibleLeftMargin, .flexibleRightMargin, .flexibleTopMargin, .flexibleBottomMargin]
        closeButton.setTitle("关闭", for: .normal)
        closeButton.setTitleColor(UIColor.white, for: .normal)
        closeButton.backgroundColor = UIColor.red
        closeButton.addTarget(self, action: #selector(closeButtonClick), for: .touchUpInside)
        closeButton.layer.masksToBounds = true
        closeButton.layer.cornerRadius = 10.0
        view.addSubview(closeButton)
    }

    @objc func closeButtonClick() {
        dismiss(animated: true, completion: nil)
    }
    
    func onAdRendered() {
        NSLog("Adtalos onAdRendered")
    }

    func onAdImpressionFinished() {
        NSLog("Adtalos onAdImpressionFinished")
    }

    func onAdImpressionReceivedError(_ error: Error!) {
        NSLog("Adtalos onAdImpressionReceivedError")
    }
    
    func onAdLoaded() {
        NSLog("Adtalos onAdLoaded")
    }

    func onAdFailed(toLoad error: Error!) {
        NSLog("Adtalos onAdImpressionReceivedError")
    }

    func onAdClicked() {
        NSLog("Adtalos onAdClicked")

    }

    func onAdLeftApplication() {
        NSLog("Adtalos onAdLeftApplication")
    }

    func  onAdOpened() {
        NSLog("Adtalos onAdOpened")
    }

    func onAdClosed() {
        NSLog("Adtalos onAdClosed")
    }
}
